src/dev/build/tasks/patch_native_modules_task.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { deleteAll, download, untar } from '../lib';
+
+const BASE_URL = `https://storage.googleapis.com/native-modules`;
+const DOWNLOAD_DIRECTORY = '.native_modules';
+
+const packages = [
+  {
+    name: 're2',
+    version: '1.10.5',
+    destinationPath: 'node_modules/re2/build/Release/',
+    shas: {
+      darwin: '0640af2971828597cdca590e3f022071d6f11d5d4f01f16d80783ff840019f33',
+      linux: '938efd0719855304b00b8a41f90eb78871d41f758ecb917c8de1db406cc8c243',
+      windows: '96d8194f1bbb2105b71330b671b202b2b46e3fa09141bd600d9c284202eda7b4',
+    },
+  },
+];
+
+async function patchModule(config, log, build, platform, pkg) {
+  const platformName = platform.getName();
+  const archiveName = `${pkg.version}-${platformName}.tar.gz`;
+  const downloadUrl = `${BASE_URL}/${pkg.name}/${archiveName}`;
+  const downloadPath = config.resolveFromRepo(DOWNLOAD_DIRECTORY, archiveName);
+  const extractedPath = build.resolvePathForPlatform(platform, pkg.destinationPath);
+  log.debug(`Patching ${pkg.name} binaries from ${downloadUrl} to ${extractedPath}`);
+
+  await deleteAll([extractedPath], log);
+  await download({
+    log,
+    url: downloadUrl,
+    destination: downloadPath,
+    sha256: pkg.shas[platformName],
+    retries: 3,
+  });
+  await untar(downloadPath, extractedPath);
```

Comment:
This fails on Linux (works on Windows and macOS). We can "fix" it if we improve the `untar` function to not try to manually unzip the file first:

```diff
  await createPromiseFromStreams([
    fs.createReadStream(source),
-   createGunzip(),
    tar.extract({
      ...extractOptions,
      cwd: destination,
    }),
   ]);
```

Reply:
I messed up the linux archive, and didn't gzip the tar... I went ahead and recreated the tar with it being gzipped, so it'll download faster and we don't have to change the shared `untar` utility (which arguably should work with gzipped or non-gzipped tars)

Reply:
Ah, that's why. Side note unrelated to this PR: `tar.extract()` also does unzipping if it detects that the tar-ball is zipped, so there's no need to manually add the extra stream in the middle. So if this was left out, the `untar` function would work as you described.

Reply:
...But then we wouldn't have detected the mistake in the archive so... 😅

Reply:
Very interesting, I wonder why we're doing this manually then... I assume it was required at some point 🙈